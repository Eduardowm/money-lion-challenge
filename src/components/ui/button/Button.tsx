import { cn } from '@/lib/helpers/tailwind'
import { PropsWithChildren } from 'react'

interface ButtonProps extends PropsWithChildren {
  className?: string
  onClick?: () => void
}

function Button({ className, onClick, children }: ButtonProps) {
  return (
    <button className={cn('text-gray-500 hover:text-gray-300', className)} onClick={onClick}>
      {children}
    </button>
  )
}
export default Button
