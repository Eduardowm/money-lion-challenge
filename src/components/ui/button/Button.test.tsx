import { fireEvent, render, screen } from '@testing-library/react'

import Button from './Button'

describe('Button', () => {
  it('renders as expected', () => {
    const { container } = render(<Button>Button Text</Button>)

    expect(container).toMatchSnapshot()
  })

  it('renders as expected with optional props', () => {
    const { container } = render(
      <Button className="w-full" onClick={() => {}}>
        Button Text
      </Button>
    )

    expect(container).toMatchSnapshot()
  })

  it('calls onClick when user clicks the button', () => {
    const mockOnClick = jest.fn()
    render(<Button onClick={mockOnClick}>Button Text</Button>)

    fireEvent.click(screen.getByText('Button Text'))

    expect(mockOnClick).toHaveBeenCalledTimes(1)
  })
})
