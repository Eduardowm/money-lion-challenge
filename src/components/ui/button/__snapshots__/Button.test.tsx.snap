// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button renders as expected 1`] = `
<div>
  <button
    class="text-gray-500 hover:text-gray-300"
  >
    Button Text
  </button>
</div>
`;

exports[`Button renders as expected with optional props 1`] = `
<div>
  <button
    class="text-gray-500 hover:text-gray-300 w-full"
  >
    Button Text
  </button>
</div>
`;
